### PR TITLE
⚠️ Move Status.APIEndpoints to Spec.ControlPlaneEndpoint

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

/hold
on https://github.com/kubernetes-sigs/cluster-api/pull/1789

### CODE REVIEW

**Comment by ncdc** [comment]
Reviewing

File: api/v1alpha2/awscluster_conversion.go [hunk 1]
```diff
@@ -46,6 +74,11 @@ func (dst *AWSClusterList) ConvertFrom(srcRaw conversion.Hub) error { // nolint
 	return Convert_v1alpha3_AWSClusterList_To_v1alpha2_AWSClusterList(src, dst, nil)
 }
 
+// Convert_v1alpha2_AWSClusterStatus_To_v1alpha3_AWSClusterStatus is an autogenerated conversion function.
```
Comment (ncdc) [inline api/v1alpha2/awscluster_conversion.go]: I think this comment needs to be updated to indicate that it's not autogenerated

Comment (vincepri) [reply api/v1alpha2/awscluster_conversion.go]: good catch

**Pull request merged**